src/main/host/syscall/handler/mod.rs

```diff
@@ -268,16 +268,14 @@ impl SyscallHandler {
                 .expect("flushing syscall ptrs");
         }
 
-        if ctx.host.shim_shmem().model_unblocked_syscall_latency
-            && ctx.process.is_running()
-            && !matches!(rv, Err(SyscallError::Blocked(_)))
-        {
+        if ctx.process.is_running() && !matches!(rv, Err(SyscallError::Blocked(_))) {
             let max_unapplied_cpu_latency = ctx.host.shim_shmem().max_unapplied_cpu_latency;
 
             // increment unblocked syscall latency, but only for non-shadow-syscalls, since the
             // latter are part of Shadow's internal plumbing; they shouldn't necessarily "consume"
             // time
-            if !is_shadow_syscall(syscall) {
+            if ctx.host.shim_shmem().model_unblocked_syscall_latency && !is_shadow_syscall(syscall)
+            {
```

Comment:
The interaction of the new `native_preemption_sim_interval` with the old `max_unapplied_cpu_latency` is not clear to me from the docs.

My understanding is that what the timer goes off, the shim increases `unapplied_cpu_latency` by `native_preemption_sim_interval` and does a yield syscall. Then here in shadow we only update the time `if unapplied_cpu_latency > max_unapplied_cpu_latency`. Otherwise we do nothing.

But the docs don't mention any relationship between `native_preemption_sim_interval` and `max_unapplied_cpu_latency`.

Reply:
Right. Added a note to the config spec.

FWIW I'd expect that normally `native_preemption_sim_interval` would be > `max_unapplied_cpu_latency`, making this moot. If it *is* less, than the unapplied cpu latency will still accumulate over multiple native timer intervals until either it exceeds `max_unapplied_cpu_latency` or control is returned to shadow and shadow schedules a different thread.

Reply:
(That reminds me I noticed that I couldn't find where unapplied cpu latency is discarded when switching threads, despite that being noted in a couple places in the docs. That's semi-orthogonal to this PR, though; it mostly means that more preemptions may happen than ideal/expected when unblocked syscall latency is enabled, which mostly means slightly worse performance)

Reply:
> (That reminds me I noticed that I couldn't find where unapplied cpu latency is discarded when switching threads, despite that being noted in a couple places in the docs. That's semi-orthogonal to this PR, though; it mostly means that more preemptions may happen than ideal/expected when unblocked syscall latency is enabled, which mostly means slightly worse performance)

Ah, I was mistaken. It is reset in `Process::continue`